coreclient/src/clients/api_clients.rs

```diff
@@ -40,10 +41,14 @@ impl ApiClients {
             .clients
             .lock()
             .map_err(|_| ApiClientsError::MutexPoisonError)?;
-        let client = clients
-            .entry(lookup_domain.clone())
-            .or_insert(ApiClient::initialize(lookup_domain)?);
-        Ok(client.clone())
+        match clients.entry(lookup_domain.clone()) {
```

Comment:
@kkohbrok I wonder why the lookup for own domain is done via the server url? Wouldn't it make sense also to use the domain here?

Reply:
I think that was another attempt at making the client work with an IP rather than a domain.

Reply:
I asked because different urls can point to the same server. For example when omitting or having the default https port 443.